applications/sdp/gpio/CMakeLists.txt

```diff
@@ -10,8 +10,12 @@ find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(emulated_gpio)
 
 sdp_assembly_generate("${CMAKE_SOURCE_DIR}/src/hrt/hrt.c")
+sdp_assembly_check("${CMAKE_SOURCE_DIR}/src/hrt/hrt.c")
+sdp_assembly_update("${CMAKE_SOURCE_DIR}/src/hrt/hrt.c")
```

Comment:
First, this call is misleading / bad naming.

It gives the impression that the newly generated assembly file will actually be updated, which is not the case.
This call will only create a build target which can be called by the user to update the assembly files.

Another problem with the approach is that all files will be updated, and not only those which differs.
Or the user cannot choose to update only a single file.

This is not a big issue when only a single file is in use, but will become a problem in future.

For now, please improve the naming.
For future, please create a ticket to addre